vfs/dir.go

```diff
@@ -514,6 +515,41 @@ func (d *Dir) _readDir() error {
 		return err
 	}
 
+	if d.vfs.Opt.BlockNormDupes { // do this only if requested, as it will have a performance hit
+		ciLock.Lock()
+		defer ciLock.Unlock()
+		ctx := context.TODO()
+		ci := fs.GetConfig(ctx)
+		if d.vfs.Opt.CaseInsensitive {
+			ci.IgnoreCaseSync = true
+		}
+
+		// sort entries such that NFD comes before NFC of same name
+		sort.Slice(entries, func(i, j int) bool {
+			if entries[i] != entries[j] && fs.DirEntryType(entries[i]) == fs.DirEntryType(entries[j]) && norm.NFC.String(entries[i].Remote()) == norm.NFC.String(entries[j].Remote()) {
+				if norm.NFD.IsNormalString(entries[i].Remote()) && !norm.NFD.IsNormalString(entries[j].Remote()) {
+					return true
+				}
+			}
+			return entries.Less(i, j)
+		})
+
+		// detect dupes, remove them from the list and log an error
+		normalizedNames := make(map[string]struct{}, entries.Len())
+		filteredEntries := make(fs.DirEntries, 0)
+		for _, e := range entries {
+			normName := fmt.Sprintf("%s-%T", operations.ApplyTransforms(ctx, e.Remote()), e) // include type to track objects and dirs separately
```

Comment:
Maybe we should have a version of ApplyTransforms we say explicitly which transforms we want - might save mucking about with contexts

Reply:
Interesting idea! I'm happy to make one. I didn't do it yet as I'm not sure how much it would buy us here, since we would still need to at least read the global config to get the `ci.IgnoreCaseSync` and `ci.NoUnicodeNormalization` values (and we'd still need some way of setting those values from the tests).

Reply:
I think it is a bit of a code smell to to having to adjust a `ctx` - we should use function parameters for passing values to functions.

From the [Go docs](https://pkg.go.dev/context) this is what contexts are for:

> Package context defines the Context type, which carries deadlines, cancellation signals, and other request-scoped values across API boundaries and between processes.

We've wedged configuration in there but I think changing it instead of making a function with more parameters is probably against the spirit of what context is for!

Reply:
Got it -- made this change.